src/reflection.jl

```diff
@@ -0,0 +1,34 @@
+import InteractiveUtils
+export @ka_code_typed
+
+function ka_code_typed(kernel, argtypes; ndrange=nothing, workgroupsize=nothing)
+    # get the iterspace and dynamic of a kernel
+    ndrange, workgroupsize, iterspace, dynamic = KernelAbstractions.launch_config(kernel, ndrange, workgroupsize)
+    # get the first block
+    block = @inbounds KernelAbstractions.blocks(iterspace)[1]
+    # get a context of the kernel based on the first block
+    ctx = KernelAbstractions.mkcontext(kernel, block, ndrange, iterspace, dynamic)
+    # reformat
+    if argtypes isa Type
+        argtypes = argtypes.parameters
+    end
+    # use code_typed
+    return InteractiveUtils.code_typed(KernelAbstractions.Cassette.overdub, (typeof(ctx), typeof(kernel.f), argtypes...))
+end
+
+
+macro ka_code_typed(ex0...)
```

Comment:
Can you add a docstring as well?

Ideally `@ka_code_typed` would take the same arguments as `code_typed`. For `code_typed` that is just `optimize=false`. As an example `@code_typed optimize=false sin(1.0)`.

Reply:
Yes I can add a doc string and look into adding the optimize argument. Is there an example of where optimize gets used when passing into code_typed()?

Reply:
`code_typed(sin, Tuple{Float64}, optimize=false)`